src/app/auth/authentication.service.spec.ts

```diff
@@ -179,6 +179,31 @@ describe('Service: Authentication service', () => {
     authenticationService.logIn(tokenJson);
   });
 
+  it('Openshift valid connection test if cluster needs to be encoded', (done) => {
+    // given
+    mockService.connections.subscribe((connection: any) => {
+      connection.mockRespond(new Response(
```

Comment:
I think the test should assert in this mockService that the connection url was encoded properly. Otherwise it's not clear it's testing what was changed. Ideally, the unit test should fail before the fix and pass after the fix.

Reply:
Also rather than typing as `any`, I think it can be typed as `MockConnection` from `'@angular/http/testing'`